modules/coinswap/types/genesis.go

```diff
@@ -14,13 +16,59 @@ func NewGenesisState(params Params, denom string) *GenesisState {
 
 // DefaultGenesisState creates a default GenesisState object
 func DefaultGenesisState() *GenesisState {
-	return NewGenesisState(DefaultParams(), sdk.DefaultBondDenom)
+	return &GenesisState{
+		Params:        DefaultParams(),
+		StandardDenom: sdk.DefaultBondDenom,
+		Sequence:      1,
+	}
 }
 
 // ValidateGenesis validates the given genesis state
 func ValidateGenesis(data GenesisState) error {
 	if err := sdk.ValidateDenom(data.StandardDenom); err != nil {
 		return err
 	}
+
+	var poolIds = make(map[string]bool, len(data.Pool))
+	var lptDenoms = make(map[string]bool, len(data.Pool))
+	var maxSequence = uint64(1)
```

Comment:
```suggestion
	var maxSequence = uint64(0)
```

If data.Pool is empty, `maxSequence != data.Sequence` is false.